.github/workflows/ci-unix-shared-installed.yml

```diff
@@ -110,6 +100,8 @@ jobs:
         -DAVIF_BUILD_EXAMPLES=ON -DAVIF_BUILD_APPS=ON
         -DAVIF_BUILD_TESTS=ON -DAVIF_ENABLE_GTEST=ON -DAVIF_LOCAL_GTEST=ON
         -DAVIF_BUILD_GDK_PIXBUF=ON -DCMAKE_INSTALL_PREFIX=./install
+        -DENABLE_AVX2=OFF
```

Comment:
A small percentage of macOS github runners don't have avx2 support but advertise that they do, causing "Illegal instruction: 4" errors in AOM tests.

Reply:
I can't come up with a way to conditionally set `-DENABLE_AVX2` for macOS in aom.cmd in a way that's also compatible with sh and cmd.exe...short of creating a separate aom.sh file. So this might need to wait until aom is built as a subdirectory.

Reply:
Is there a way to factor this out into a separate small PR? These "ILLEGAL" errors are really annoying. I tried creating a PR that just adds these two lines to the yml files but it seems that ENABLE_AVX2 is not recognized by cmake. I guess it's tied to other changes in this PR?

And a few more questions on this topic:
- This sounds like an issue with github runners that we're working around. Do you know if there's an issue open somewhere?
- Why is the CMAKE_OSX_DEPLOYMENT_TARGET flag needed?
- In practice, I'm fairly sure I've only ever seen the ci-unix-static one fail. I doubt it's just a fluke. Maybe we don't need the changes in the other two workflows?

Thanks!

Reply:
The `ENABLE_AVX2` flag is actually in AOM (see [here](https://aomedia.googlesource.com/aom/+/e2318cdca95519318af93205fe6d9156b5353924/build/cmake/aom_config_defaults.cmake#53)). In this pull request, I am adding libaom with `add_subdirectory` to incorporate it into the larger cmake project, which allows one to pass `-DENABLE_AVX2=OFF`. Whereas on the libavif main branch it gets built in a separate step by running aom.cmd. I can't think of a way to pass arguments to that script that would work for windows and unix, and we presumably don't want to disable avx2 for everyone just to get around this issue.

As for your observation: I _have_ seen this error in all of the workflows that use macos, but it does seem to be much more frequent for ci-unix-static. And there is a pattern I've noticed: the builds seem to fail for macOS 12.7 but not for 12.6.9 (you can see the version by expanding the "Operating system" output in the "Set up job" step). If you use tmate to open a shell and run `sysctl machdep.cpu`, you'll see that a 12.6.9 runner always has AVX2 and a 12.7 runner does not. Anecdotally it seems like the ci-unix-static gets assigned the 12.7 worker more often. I can only speculate as to why that is.

I found an issue opened a year ago to runner-images on the subject of AVX2 in macos runners: https://github.com/actions/runner-images/issues/6845. The user who closed that issue references a comment on a separate issue https://github.com/actions/runner-images/issues/5609#issuecomment-1137149036, which reads:

> Unfortunately, at this time out agents are not guaranteed to have AVX2 CPU instruction, and only the way to enable the feature for you on permanent basis is to use self-hosted agents(where it is possible to control the underlying hardware).

I don't know if it's worth raising a new issue, since it seems like the 12.6.9 runners have AVX2 pretty reliably, and the 12.7 runners represent a regression of sorts. 

FWIW, AOM actually does cpu feature detection before it uses its own avx2 optimizations, but when building it will add the `-mavx2` compile flag so long as the compiler supports it, and that enables avx2 optimizations in glibc. The failures in these tests all seem to be caused by a call to the `round` function, which has a `vpbroadcastq` instruction in the SIMD-optimized version.

Reply:
The CMAKE_OSX_TARGET is a red herring. Before I figured out that the hardware was to blame, I had hoped that targeting a specific OS X version might resolve this apparently discrepancy between 12.6.9 and 12.7.

Reply:
Thanks for all the extra context, I understand the issue much better now.
Can we maybe use a `sed` command to add  `-DENABLE_AVX2=OFF` to aom.cmd in the CI workflows, similar to what we used to do for shared libs here? https://github.com/AOMediaCodec/libavif/commit/12e066686892df1c8201cfb0d8d6c68ad248c872#diff-cf7515a55275bc6077268c78cd03a0f56422f2e21e35040f907cb7965aed4f14L45
It's not *pretty* but I'll take any fix at this point.

Reply:
yes, I think that ought to work.

Reply:
Just make sure to do it before the actions/cache call